### PR TITLE
Standardize 2D->2d and 3D->3d.

### PR DESCRIPTION
**Pull request by bangerth**
Seen while looking at #5041. We have a funny mix between 2d/3d and 2D/3D:
```
projects/aspect> egrep -r '\".*[23]d' include source/ | wc -l
123
projects/aspect> egrep -r '\".*[23]D' include source/ | wc -l 
82
```
So the lower-case `d` wins statistically speaking, and this patch fixes this uniformly in strings, almost all of which appear to be in the descriptions of parameters and models. Let's see whether this compiles and tests alright.

/rebuild

### CODE REVIEW

File: source/structured_data.cc [hunk 1]
```diff
@@ -1478,24 +1478,24 @@ namespace aspect
 
       prm.enter_subsection (subsection_name);
       {
-        prm.declare_entry("Slice dataset in 2D plane", "false",
+        prm.declare_entry("Slice dataset in 2d plane", "false",
```
Comment (tjhei) [inline source/structured_data.cc:1481]: This will break things.

File: source/structured_data.cc [hunk 2]
```diff
@@ -1516,7 +1516,7 @@ namespace aspect
 
       prm.enter_subsection(subsection_name);
       {
-        slice_data = prm.get_bool ("Slice dataset in 2D plane");
+        slice_data = prm.get_bool ("Slice dataset in 2d plane");
```
Comment (tjhei) [inline source/structured_data.cc:1519]: Same here

**Review by tjhei — CHANGES_REQUESTED** [review]

**Comment by bangerth** [comment]
Oh, doh -- it hadn't occurred to me that there could be places where that appears in the names of parameters. I reverted the two places, thanks for pointing it out!

**Comment by tjhei** [comment]
> , thanks for pointing it out

The testers did. :-)

**Review by tjhei — APPROVED** [review]

**Pull request merged**